src/data-structures/bloomfilter.js

```diff
@@ -0,0 +1,187 @@
+/**
+ * Bloomfilter.
+ *
+ * @example
+ *
+ * @module data-structures/bloomfilter
+ */
+(function(exports) {
+  'use strict';
+
+  function randomUint32() {
+    return Math.floor(Math.random() * Math.pow(2, 32));
+  }
+  /**
+   * Calculate a 32 bit FNV-1a hash
+   * Found here: https://gist.github.com/vaiorabbit/5657561
+   * Ref.: http://isthe.com/chongo/tech/comp/fnv/
+   *
+   * @param {string} str the input value
+   * @param {boolean} [asString=false] set to true to return the hash value as
+   *     8-digit hex string instead of an integer
+   * @param {integer} [seed] optionally pass the hash of the previous chunk
+   * @returns {integer | string}
+   */
+  function hashFnv32a(str, asString, seed) {
+    /*jshint bitwise:false */
+    var i;
+    var l;
+    var hval = seed === undefined ? 0x811c9dc5 : seed;
+
+    for (i = 0, l = str.length; i < l; i = i + 1) {
+      hval ^= str.charCodeAt(i);
+      hval +=
+        (hval << 1) + (hval << 4) + (hval << 7) + (hval << 8) + (hval << 24);
+    }
+    if (asString) {
+      // Convert to 8 digit hex string
+      return ('0000000' + (hval >>> 0).toString(16)).substr(-8);
+    }
+    return hval >>> 0;
+  }
+
+  // make a hash function
+  function mkHashFun(seed, limit) {
+    return function(value) {
+      return hashFnv32a(value, false, seed) % limit;
+    };
+  }
+  /**
+   * Create a bitmap with a given size
+   * Bit map is not resizeable
+   * @public
+   * @constructor
+   * @param {Number} size the size of the bitmap
+   */
+  exports.Bitmap = function(size) {
+    size = size || 1024;
+    if (size < 0) {
+      throw 'the size cannot be negative';
```

Comment:
Let's use `Error` so the user can get a stack trace. Also we can start with a capital letter.